recurring_contract/model/recurring_contract.py

```diff
@@ -291,6 +326,20 @@ def clean_invoices(self, cr, uid, ids, context=None, since_date=None,
             wf_service.trg_validate(uid, 'account.invoice',
                                     inv.id, 'invoice_open', cr)
 
+        for contract in self.browse(cr, uid, ids, context):
+            next_invoice_date = since_date
+
+            if contract.last_paid_invoice_date:
+                next_invoice_date = max(
+                    datetime.strptime(
+                        contract.last_paid_invoice_date, DF), since_date)
+
+            self.write(

```

Comment:
I think you should call super write method to avoid having an error when the next_invoice_date is rewinded.